testsuite failure against gettext 0.21.1

after upgrading gettext from 0.21 to 0.21.1, the testsuite fails with: 

```
[   24s] =================================== FAILURES ===================================
[   24s] ________________________ TestCPOFile.test_wrap_gettext _________________________
[   24s] 
[   24s] self = <translate.storage.test_cpo.TestCPOFile object at 0x7fdb7e6118b0>
[   24s] 
[   24s]         def test_wrap_gettext(self):
[   24s]             posource = r"""# Test
[   24s]     msgid ""
[   24s]     msgstr ""
[   24s]     "Project-Id-Version: kmail\n"
[   24s]     "POT-Creation-Date: 2020-05-11 04:03+0200\n"
[   24s]     "PO-Revision-Date: 2020-05-12 14:13+0000\n"
[   24s]     "Last-Translator: Roman Savochenko <roman@oscada.org>\n"
[   24s]     "Language-Team: Ukrainian <https://mirror.git.trinitydesktop.org/weblate/"
[   24s]     "projects/tdepim/kmail/uk/>\n"
[   24s]     "Language: uk\n"
[   24s]     "MIME-Version: 1.0\n"
[   24s]     "Content-Type: text/plain; charset=UTF-8\n"
[   24s]     "Content-Transfer-Encoding: 8bit\n"
[   24s]     "Plural-Forms: nplurals=3; plural=n%10==1 && n%100!=11 ? 0 : n%10>=2 && n"
[   24s]     "%10<=4 && (n%100<10 || n%100>=20) ? 1 : 2;\n"
[   24s]     "X-Generator: Weblate 4.0.4\n"
[   24s]     
[   24s]     #: configuredialog.cpp:4580
[   24s]     msgid ""
[   24s]     "x: to be continued with \"do not loop\", \"loop in current folder\", and "
[   24s]     "\"loop in all folders\".\n"
[   24s]     "When trying to find unread messages:"
[   24s]     msgstr "При спробі знайти не прочитані повідомлення:"
[   24s]     
[   24s]     msgid ""
[   24s]     "You can get a copy of your Recovery Key by going to &syncBrand.shortName."
[   24s]     "label; Options on your other device, and selecting  \"My Recovery Key\" "
[   24s]     "under \"Manage Account\"."
[   24s]     msgstr ""
[   24s]     """
[   24s] >           assert self.poreflow(posource) == posource
[   24s] E           assert ('# Test\n'\n 'msgid ""\n'\n 'msgstr ""\n'\n '"Project-Id-Version: kmail\\n"\n'\n '"POT-Creation-Date: 2020-05-11 04:03+0200\\n"\n'\n '"PO-Revision-Date: 2020-05-12 14:13+0000\\n"\n'\n '"Last-Translator: Roman Savochenko <roman@oscada.org>\\n"\n'\n '"Language-Team: Ukrainian <https://mirror.git.trinitydesktop.org/weblate/"\n'\n '"projects/tdepim/kmail/uk/>\\n"\n'\n '"Language: uk\\n"\n'\n '"MIME-Version: 1.0\\n"\n'\n '"Content-Type: text/plain; charset=UTF-8\\n"\n'\n '"Content-Transfer-Encoding: 8bit\\n"\n'\n '"Plural-Forms: nplurals=3; plural=n%10==1 && n%100!=11 ? 0 : n%10>=2 && "\n'\n '"n%10<=4 && (n%100<10 || n%100>=20) ? 1 : 2;\\n"\n'\n '"X-Generator: Weblate 4.0.4\\n"\n'\n '\n'\n '#: configuredialog.cpp:4580\n'\n 'msgid ""\n'\n '"x: to be continued with \\"do not loop\\", \\"loop in current folder\\", '\n 'and "\n'\n '"\\"loop in all folders\\".\\n"\n'\n '"When trying to find unread messages:"\n'\n 'msgstr "При спробі знайти не прочитані повідомлення:"\n'\n '\n'\n 'msgid ""\n'\n '"You can get a copy of your Recovery Key by going to &syncBrand.shortName."\n'\n '"label; Options on your other device, and selecting  \\"My Recovery Key\\" '\n '"\n'\n '"under \\"Manage Account\\"."\n'\n 'msgstr ""\n') == ('# Test\n'\n 'msgid ""\n'\n 'msgstr ""\n'\n '"Project-Id-Version: kmail\\n"\n'\n '"POT-Creation-Date: 2020-05-11 04:03+0200\\n"\n'\n '"PO-Revision-Date: 2020-05-12 14:13+0000\\n"\n'\n '"Last-Translator: Roman Savochenko <roman@oscada.org>\\n"\n'\n '"Language-Team: Ukrainian <https://mirror.git.trinitydesktop.org/weblate/"\n'\n '"projects/tdepim/kmail/uk/>\\n"\n'\n '"Language: uk\\n"\n'\n '"MIME-Version: 1.0\\n"\n'\n '"Content-Type: text/plain; charset=UTF-8\\n"\n'\n '"Content-Transfer-Encoding: 8bit\\n"\n'\n '"Plural-Forms: nplurals=3; plural=n%10==1 && n%100!=11 ? 0 : n%10>=2 && n"\n'\n '"%10<=4 && (n%100<10 || n%100>=20) ? 1 : 2;\\n"\n'\n '"X-Generator: Weblate 4.0.4\\n"\n'\n '\n'\n '#: configuredialog.cpp:4580\n'\n 'msgid ""\n'\n '"x: to be continued with \\"do not loop\\", \\"loop in current folder\\", '\n 'and "\n'\n '"\\"loop in all folders\\".\\n"\n'\n '"When trying to find unread messages:"\n'\n 'msgstr "При спробі знайти не прочитані повідомлення:"\n'\n '\n'\n 'msgid ""\n'\n '"You can get a copy of your Recovery Key by going to &syncBrand.shortName."\n'\n '"label; Options on your other device, and selecting  \\"My Recovery Key\\" '\n '"\n'\n '"under \\"Manage Account\\"."\n'\n 'msgstr ""\n')
[   24s] E               # Test
[   24s] E               msgid ""
[   24s] E               msgstr ""
[   24s] E               "Project-Id-Version: kmail\n"
[   24s] E               "POT-Creation-Date: 2020-05-11 04:03+0200\n"
[   24s] E               "PO-Revision-Date: 2020-05-12 14:13+0000\n"
[   24s] E               "Last-Translator: Roman Savochenko <roman@oscada.org>\n"
[   24s] E               "Language-Team: Ukrainian <https://mirror.git.trinitydesktop.org/weblate/"
[   24s] E               "projects/tdepim/kmail/uk/>\n"
[   24s] E               "Language: uk\n"
[   24s] E               "MIME-Version: 1.0\n"
[   24s] E               "Content-Type: text/plain; charset=UTF-8\n"
[   24s] E               "Content-Transfer-Encoding: 8bit\n"
[   24s] E             - "Plural-Forms: nplurals=3; plural=n%10==1 && n%100!=11 ? 0 : n%10>=2 && n"
[   24s] E             ?                                                                         -
[   24s] E             + "Plural-Forms: nplurals=3; plural=n%10==1 && n%100!=11 ? 0 : n%10>=2 && "
[   24s] E             - "%10<=4 && (n%100<10 || n%100>=20) ? 1 : 2;\n"
[   24s] E             + "n%10<=4 && (n%100<10 || n%100>=20) ? 1 : 2;\n"
[   24s] E             ?  +
[   24s] E               "X-Generator: Weblate 4.0.4\n"
[   24s] E               
[   24s] E               #: configuredialog.cpp:4580
[   24s] E               msgid ""
[   24s] E               "x: to be continued with \"do not loop\", \"loop in current folder\", and "
[   24s] E               "\"loop in all folders\".\n"
[   24s] E               "When trying to find unread messages:"
[   24s] E               msgstr "При спробі знайти не прочитані повідомлення:"
[   24s] E               
[   24s] E               msgid ""
[   24s] E               "You can get a copy of your Recovery Key by going to &syncBrand.shortName."
[   24s] E               "label; Options on your other device, and selecting  \"My Recovery Key\" "
[   24s] E               "under \"Manage Account\"."
[   24s] E               msgstr ""
````

its not entirely clear to me whether this is an issue with the testsuite, with translate or with gettext. please help.

--- a/translate/storage/pypo.py
+++ b/translate/storage/pypo.py
@@ -142,7 +142,6 @@ class PoWrapper(textwrap.TextWrapper):
             \s+|                                  # any whitespace
             [a-z0-9A-Z_-]+/|                      # nicely split long URLs
             \w*\\.\w*|                            # any escape should not be split
-            n(?=%)|                               # wrap insidide plural equation
             \.(?=\w)|                             # full stop inside word
             [\w\!\'\&\.\,\?=<>%]+\s+|             # space should go with a word
             [^\s\w]*\w+[a-zA-Z]-(?=\w+[a-zA-Z])|  # hyphenated words

--- a/translate/storage/test_po.py
+++ b/translate/storage/test_po.py
@@ -1064,8 +1064,8 @@ msgstr ""
 "MIME-Version: 1.0\n"
 "Content-Type: text/plain; charset=UTF-8\n"
 "Content-Transfer-Encoding: 8bit\n"
-"Plural-Forms: nplurals=3; plural=n%10==1 && n%100!=11 ? 0 : n%10>=2 && n"
-"%10<=4 && (n%100<10 || n%100>=20) ? 1 : 2;\n"
+"Plural-Forms: nplurals=3; plural=n%10==1 && n%100!=11 ? 0 : n%10>=2 && "
+"n%10<=4 && (n%100<10 || n%100>=20) ? 1 : 2;\n"
 "X-Generator: Weblate 4.0.4\n"
 
 #: configuredialog.cpp:4580
@@ -1081,7 +1081,13 @@ msgid ""
 "under \"Manage Account\"."
 msgstr ""
 """
-        assert self.poreflow(posource) == posource
+        reflowed = self.poreflow(posource)
+        if reflowed == posource:
+            # Gettext 0.21.1 and newer
+            assert self.poreflow(posource) == posource
+        else:
+            # Gettext 0.21 and older
+            assert self.poreflow(posource) == posource.replace('"\n"n%10', 'n"\n"%10')
 
     def test_msgidcomments(self):
         posource = r"""

